clippy_lints/src/casts/mod.rs

```diff
@@ -371,12 +373,13 @@ declare_clippy_lint! {
 
 declare_clippy_lint! {
     /// ### What it does
-    /// Checks for `as` casts between raw pointers without changing its mutability,
-    /// namely `*const T` to `*const U` and `*mut T` to `*mut U`.
+    /// Checks for `as` casts on a raw pointer that don't change its
+    /// mutability, namely `*const T` to `*const U` and `*mut T` to `*mut U`.
```

Comment:
I find "between" to work better to describe casts, since we're talking about the before & after types "on" feels a bit ambiguous

We use mutability here and constness elsewhere to describe the same thing, it would be good to stick with one

Reply:
Good idea!